Analytics-CSharp/Segment/Analytics/Analytics.cs

```diff
@@ -138,14 +125,14 @@ public JsonObject Traits()
         /// </summary>
         public void Reset()
         {
-            userInfo.userId = null;
-            userInfo.anonymousId = null;
-            userInfo.traits = null;
+            this._userInfo._userId = null;
+            this._userInfo._anonymousId = null;
+            this._userInfo._traits = null;
```

Comment:
I forgot about this last time. it's probably better that we just create a new UserInfo here, instead of assigning the properties one by one. also, assigning anonymousId as null actually introduce a bug, since the `ResetAction` below actually gives a random uuid, which creates a inconsistent image between memory and sovran. 
refer to the kotlin counterpart [here](https://github.com/segmentio/analytics-kotlin/pull/124/files#diff-4afefc77facdcb9004dce53cca6171b811fa2c5b1b9908c23bbbd5d9a0098eeaR534) for solution.

Reply:
Fixed.